core/src/CMakeLists.txt

```diff
@@ -1,15 +1,29 @@
 #I have to leave these here for tribits
-KOKKOS_INCLUDE_DIRECTORIES(
-  ${CMAKE_CURRENT_BINARY_DIR}
-  ${CMAKE_CURRENT_SOURCE_DIR}
-  ${KOKKOS_TOP_BUILD_DIR}
-)
+IF (KOKKOS_ENABLE_IMPL_DESUL_ATOMICS)
+  find_package(desul_atomics QUIET)
```

Comment:
I think it would be better to have a flag, `KOKKOS_IMPL_USE_DESUL_EXTERNAL_LIB` or something like that, then make the `find_package` `REQUIRED`. If not you can get some weird issues where if you make a configuration mistake your code will still compile but with potentially the wrong version of desul atomics

Reply:
> I agree with @masterleinad in that we should treat it as a TPL if it can be external. Minor question: would it make sense to include it as a subproject (i.e. add_subdirectory'd)?

the subdirectory stuff doesn't make much sense, because desul is pulling in BLT which is like 100k of cmake with its own random options for CUDA and what not. So we would need to translate that all.